### DISCUSSION TITLE
automatic workflows for service contracts

### DISCUSSION THREAD

**Discussion by krseguros** [YetiForceCompany/YetiForceCRM#16939] [How to]
Greetings,

I can't activate automatic workflows for service contracts, [CRON] is running correctly.

I am looking to activate automatic workflows for service contracts so that an activity is created whenever service contracts have an End Date in 15 days or less.

I have tried several ways and options in the flow rule but the activity is not created automatically.
![image](https://user-images.githubusercontent.com/96985476/222857934-2d9dc5d4-cf48-4e62-a94f-aa7c8dbb084b.png)

![image](https://user-images.githubusercontent.com/96985476/222858058-951bdf09-8b31-44df-b3e5-f03bc218ff38.png)

![image](https://user-images.githubusercontent.com/96985476/222858131-a954a505-6e9f-4be8-bea1-367b7842bf1d.png)

**Comment by krseguros** [comment]
Hello,

Does anyone have an answer to this?

**Reply by krseguros** [reply]
.

**Comment by mariuszkrzaczkowski** [comment]
give an example

**Comment by krseguros** [comment]
Example:

In the Service contracts module I have the contract "X" in this module I have the date field "End of Validity Date", this contract "X" has an end date of 03-15-2023, using the existing mechanisms I try to create a flow of work that creates an activity (task/event) related to this contract "X" 15 days before its "End of Validity Date", this in order that the user can manage and place actions for the renewal of said contract.
I need this flow to be executed automatically using as a parameter that the contract is current and that "End of Validity Date" is less than 15 days, the status of the contract and the "End of Validity Date" is placed by the user when creating said contract.

**Comment by mariuszkrzaczkowski** [comment]
try this configuration
![image](https://user-images.githubusercontent.com/10198654/223346755-6eaab676-6612-4319-b454-a020ce32a317.png)
![image](https://user-images.githubusercontent.com/10198654/223346773-f92bdf15-e0a1-4d6a-b778-45cccedde3b6.png)

**Reply by krseguros** [reply]
Thanks @mariuszkrzaczkowski , the tasks/activities are already being created, however, I would like to know why I can't see the history of these activities? For example, the contract "x" has 3 related activities, however, I can only see one, the one that is in delayed status
![image](https://user-images.githubusercontent.com/96985476/224421361-6a968e84-f41d-4c57-bc46-958a4b8f4abb.png)
![image](https://user-images.githubusercontent.com/96985476/224421534-ec58b033-7524-407d-bf75-16ff9c383f1d.png)
